Fix issue DPTP-4032 Add instructions on accessing RedHat intranet (#490)
openshift/ci-docs: content/en/docs/architecture/ci-operator.md

the [hypershift-hostedcluster-workflow](https://steps.ci.openshift.org/workflow/hypershift-hostedcluster-workflow)
which can be achieved simply by renaming the name of the workflow.

### Allowing Tests to Access RedHat Intranet
{{% alert title="LIMITED SUPPORT" color="warning" %}}

> `Accessing RedHat Intranet` is a pilot project, and the support from the Test Platform is limited.

Reviewer: ```suggestion
`Accessing RedHat Intranet` is a pilot project, and support from the Test Platform is limited.
```

> `Accessing RedHat Intranet` is a pilot project, and the support from the Test Platform is limited.

Reviewer 2: ```suggestion
`Accessing Red Hat Intranet` is a pilot project, and the support from the Test Platform is limited.
```
Red Hat is always two words (https://www.redhat.com/en/about/brand/standards/naming-and-trademarks)

{{% /alert %}}

To enable tests to access the RedHat intranet, the `restrict_network_access` flag must be set to `false` within the test section.

```yaml
- as: my-e2e
  commands: curl -XPOST https://devservices.dpp.openshift.com/support/general_request
  from: stable:cli
  resources:
    requests:
      cpu: 100m
      memory: 200Mi
  restrict_network_access: false # explicitly allow the test to access the RedHat intranet
```

> There are two levels of network access control for accessing the RedHat intranet: 1. `restrict_network_access` flag and 2. network ACL maintained by the IT.

Reviewer: ```suggestion
There are two levels of network access control for accessing the RedHat intranet: the `restrict_network_access` flag, and network ACL maintained by IT.
```
